lib/spider/Dockerfile

```diff
@@ -0,0 +1,135 @@
+# Cuda-based image which installs a Python Poetry environment designed to be used interactively.
+#
+# Depending on the machine configuration, you might need to add `sudo` at the beginning of all the 
+# `docker` commands below. 
+#
+# To build this image, use:
+#   docker build . \
+#       --tag <image-name> \
+#       --build-arg PROJECT_NAME=<project-name>
+# To customize it, you can use the following example:
+#   docker build . \
+#       --tag <image-name> \
+#       --build-arg PROJECT_NAME=<project-name> \
+#       --build-arg ALIAS_USERNAME=<username> \
+#       --build-arg ALIAS_REPOSITORY=dotfiles \
+#       --build-arg ALIAS_PATH=.config/oh-my-zsh \
+#       --build-arg NEOVIM_USERNAME=<username> \
+#       --build-arg NEOVIM_DEPS_INSTALL_COMMAND="apt install -y npm zip" \
+#       --build-arg NEOVIM_TOOLS_INSTALL_COMMAND="+MasonInstallAll"
+# If this has been run with another configuration, you might need to destroy the cache by adding
+# `--no-cache` to introduce a new configuration.
+#
+# To use this image interactively, use:
+#   docker run -it --rm --gpus all \    
+#       --volume $(pwd):/opt/pysetup/ \
+#       <image-name> 
+# To share some local cache with the container, for instance HuggingFace's models, use:
+#   docker run -it --rm --gpus all \    
+#       --volume $(pwd):/opt/pysetup/ \
+#       --volum ~/.cache/huggingface:/root/.cache/huggingface \
+#       <image-name> 
+
+FROM nvidia/cuda:12.2.0-devel-ubuntu22.04
+
+ARG PROJECT_NAME
+
+ARG PYTHON_VERSION=3.11
+ARG POETRY_VERSION=1.6.1
+ARG ALIAS_USERNAME=ohmyzsh
+ARG ALIAS_REPOSITORY=ohmyzsh
+ARG ALIAS_PATH=custom
+ARG NEOVIM_USERNAME=nvim-lua
+ARG NEOVIM_REPOSITORY=kickstart.nvim
+ARG NEOVIM_DEPS_INSTALL_COMMAND=""
+ARG NEOVIM_PLUGIN_INSTALL_COMMAND="'+Lazy install'"
+ARG NEOVIM_TOOLS_INSTALL_COMMAND=""
+
+    # python
+ENV PYTHON_VERSION=$PYTHON_VERSION \
+    \
+    # poetry
+    # https://python-poetry.org/docs/configuration/#using-environment-variables
+    POETRY_VERSION=$POETRY_VERSION \
+    # make poetry install to this location
+    POETRY_HOME="/opt/poetry" \
+    # make poetry NOT create the virtual environment in the project's root, as it would be 
+    # overwriten when setting a volume on /opt/pysetup/
+    POETRY_VIRTUALENVS_IN_PROJECT=false \
+    # do not ask any interactive question
+    POETRY_NO_INTERACTION=1 \
+    \
+    # paths
+    CONFIG_PATH="/root/.config" \
+    SETUP_PATH="/opt/setup" \
+    PYSETUP_PATH="/opt/pysetup" 
+
+# prepend poetry to path
+ENV PATH="$POETRY_HOME/bin:$PATH"
```

Comment:
To make sure i understand correctly, this line is needed so we can call the command`poetry install` anywhere in the docker?

Reply:
I think so, I'm not 100% sure, I didn't check what's happening if we remove this! This comes directly from the Dockerfile we took as basis when creating the multi-stage build dockerfile for Poetry, since we took it as a ground truth to start from, we didn't challenge all of its content